exercises/hangman/src/test/java/HangmanTest.java

```diff
@@ -0,0 +1,222 @@
+
+        // Emitters respectively for the word and letter observables
+        ObservableEmitter<String>[] emitters = new ObservableEmitter[2];
+        Runnable emit = () -> {
+            // Process sending the inputs in the right order
+            emitters[0].onNext("secret");
+            Stream.of("a", "e", "o", "s", "c", "r", "g", "t").forEach(emitters[1]::onNext);
+            emitters[0].onNext("abba");
+            Stream.of("a", "e", "s", "b").forEach(emitters[1]::onNext);
+            emitters[0].onComplete();
+        };
+        Observable<String> words = createWordObservable(emitters, emit);
+        Observable<String> letters = createLetterObservable(emitters, emit);
+        Observable<Output> outputs = hangman.play(words, letters);
+
+        // We collect the results of the game
+        List<Output> results = new ArrayList<>();
+        Disposable subscription = outputs.filter(output -> output.status != Status.PLAYING)
+            .subscribe(results::add);
+        try {
+            assertEquals(2, results.size());
+
+            Output first = results.get(0);
+            assertEquals("secret", first.discovered);
+            assertEquals(setOf("s", "e", "c", "r", "t"), first.guess);
+            assertEquals(setOf("a", "o", "g"), first.misses);
+            assertEquals(Status.WIN, first.status);
+
+            Output second = results.get(1);
+            assertEquals("abba", second.discovered);
+            assertEquals(setOf("a", "b"), second.guess);
+            assertEquals(setOf("e", "s"), second.misses);
+            assertEquals(Status.WIN, first.status);
+        } finally {
+            subscription.dispose();
+        }
+    }
+
+    Observable createWordObservable(ObservableEmitter[] emitters, Runnable emit) {
+        return Observable.create(
+            emitter -> {
+                // A new subscription was created for words, record it.
+                emitters[0] = emitter;
+                if (emitters[1] != null) {
+                    // Start emitting only when both word and letter observable have subscriptions
+                    emit.run();
+                }
+            });
+    }
+
+    Observable createLetterObservable(ObservableEmitter[] emitters, Runnable emit) {
+        return Observable.create(
+            emitter -> {
+                // A new subscription was created for letters, record it.
+                emitters[1] = emitter;
+                if (emitters[0] != null) {
+                    // Start emitting only when both word and letter observable have subscriptions
+                    emit.run();
+                }
+            });
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void cannotPlayAGuessTwice() {
+        Observable<Output> result = hangman.play(
+            Observable.fromArray("secret"),
+            Observable.fromArray("e", "c", "s", "c"));
+
+        expectedException.expect(IllegalArgumentException.class);
```

Comment:
Use assertThrows instead.

Also, optionally use AssertJ to improve the assertions on the returned IAE.

Reply:
I do like assertj. Because I haven't seen it in any project, I stayed with JUnit. I'll add ait AND move to JUnit 4.13 to stay up to date.